### ISSUE TITLE
Report VHDL Reserved Words

### ISSUE THREAD

**Issue by SittingDuc** [jeremiah-c-leary/vhdl-style-guide#1053]
The VHDL LRM has a short list (I count 120) of reserved keywords that identifiers (signals, functions, labels, etc) cannot have - "signal", "port", "bus", "default", etc. Each new revision generally adds a few new words - "protected", "private", "context" etc. (I also count 4 words removed in the history of the standard)

At present nothing in vsg nor vscode alerts me that a given identifier has been given a reserved name, which is complicating my use-case - "make this VHDL-1993-ish code compile cleanly under VHDL-2008". For now I have a workaround of trying to compile the code in ModelSim, which is pretty good about protesting about the new keywords _that it knows about_, but a linter message would be nice too.

A useful feature would be a rule that checks each identifier is not on the set of reserved words. For myself, any word reserved from 1987 to 2019 is sufficient; but other users might appreciate "only test words from VHDL-2002" if that is the standard their project is being compiled to.

It is also worth noting that functions can sometimes be called operations - such as "nand", "xor", when overloading the defined operations; but that naming a signal "xor" is more likely an error.

Thanks,

**Comment by SittingDuc** [comment]
An example will help. The attached file passes vsg with no reported issues, despite almost every identifier in it being a reserved keyword

[keyword_test.vhd.txt](https://github.com/jeremiah-c-leary/vhdl-style-guide/files/13482754/keyword_test.vhd.txt)

With the proposed feature request, the test file will report a count of issues (8-14 depending on what is fair to trip a message)

Thanks again,

**Comment by SittingDuc** [comment]
Hello,

I grabbed a copy of the git tree on the branch `issue-1053` and ran it over some files, real and artificial and it works, so the new test is good by me.

Reading over the code, I did note some symbols got duplicated in `dMap['2008']` (file rule_001.py at or about line 26), `'assert', 'assume', 'assume_guarantee', 'cover', 'default', 'fairness', 'property', 'restrict', 'restrict_guarantee', 'sequence', 'strong', 'vmode', 'vprop', 'vunit'`... looks like a copy-pasta

I have no objections to a merge. Thank you for your work on supporting my feature request! :)
